source/agora/node/BlockStorage.d

```diff
@@ -961,14 +954,16 @@ public class MemBlockStorage : IBlockStorage
     public override void updateBlockSig (in Height height, in Hash hash,
         in Signature sig, in BitMask validators) @safe
     {
-        if (this.blocks.length < height)
-            throw new Exception("No such block");
+        ensure(this.blocks.length >= height,
+                "Signature update: No block at height {} ({})", height, this.blocks.length);
```

Comment:
```suggestion
                "Signature update: Can not update block at height {} as current height is {}.", height, this.blocks.length);
```

Reply:
"Can not update block signature at height [...]", dropping the "Signature update" ? Otherwise the "update" is repeated twice.

Reply:
Yes even better.